### ISSUE TITLE
Last.fm scrobbling [$45]

### ISSUE THREAD

**Issue by pdaddyo** [pdaddyo/soundbounce#85]
##

**Comment by danhumphries89** [comment]
I would love to see this on OSX soon, +1 here too :+1:

**Comment by danielsamuels** [comment]
:+1:

**Comment by gucklar** [comment]
As user, I think this enhancement has high priority, because last.fm users don't like players that can't scrobble. I'm not a developer, but maybe the scrobbling does not necessarily be made trough spotify: players like winamp and many more can srobble by itself...

**Comment by bgod** [comment]
I know I'll use the app a lot more once scrobbling is supported.  Maybe it's time to put a bounty on it? 

https://www.bountysource.com/issues/7139434-last-fm-scrobbling?utm_campaign=plugin&utm_content=tracker%2F9578624&utm_medium=issues&utm_source=github

**Comment by DJFliX** [comment]
I just added to the bounty because I discovered SoundBounce and I think it would be an AWESOME addition to have Last.fm scrobbling. Until it's there I'll just keep discovering new tracks using SoundBounce, but I'll listen to them as much as possible through the original client to make the scribbles happen...

**Comment by DJFliX** [comment]
I'm a Mac user

**Comment by ikt** [comment]
>.>

**Comment by USA-RedDragon** [comment]
Is this issue still applicable?

**Comment by andrewthecoder** [comment]
Yes, still applicable!

**Comment by USA-RedDragon** [comment]
How would the dev's like to handle api key required for this to work?

**Comment by pdaddyo** [comment]
This is quite easy to implement if we read the last.fm details from a config file (literally like 2 lines of code in the C# build, code would sit here: 

https://github.com/pdaddyo/soundbounce/blob/master/src/SoundBounce.Spotify.API/Session.cs#L283

However, this would require releasing a new native client, something we've not had to do since launch.  This is why it's not been done yet, because we don't have a system in place for automatic native client updates.  Also not sure if node's spotify library (as used on linux / osx) supports last.fm scrobbling at all.

So this is why I decided to leave this until the new desktop Spotify SDK drops (still awaiting word from Spotify on this - libspotify that we are currently using has been officially deprecated).   FYI The bounty isn't claimable unless it's implemented on OSX + Linux too, that's where the focus would need to be since it's so easy to implement on Windows.

**Comment by USA-RedDragon** [comment]
https://github.com/pdaddyo/soundbounce/tree/master/src/client-node-webkit/node-spotify-master
This is the library for OSX/Linux?
And I work better on those platoforms, I hate c# lol

**Comment by USA-RedDragon** [comment]
Or is it this: https://github.com/pdaddyo/node-libspotify

**Comment by pdaddyo** [comment]
Yes the latter, this one https://github.com/pdaddyo/soundbounce/tree/master/src/client-node-webkit/node-spotify-master

**Comment by USA-RedDragon** [comment]
OK, I think I got this :+1:

**Comment by USA-RedDragon** [comment]
For Linux, and OSX can we either include or have them install liblastfm? It's officially open sourced and maintained by lastfm, and there's no use in rewriting a perfectly fine library.

**Comment by pdaddyo** [comment]
Scrobbling is supported natively by libspotify which is already included (simply by supplying last.fm credentials at login), see here https://developer.spotify.com/docs/libspotify/12.1.51/group__session.html#ga2475a1fbc2eeacccb96b7f36efd36725  - so I'd rather not include another dependency if possible.
